src/symmetric.rs

```diff
@@ -471,6 +643,94 @@ impl SymmetricAlgorithmKey {
     }
 }
 
+#[cfg(feature = "block-cipher-trait")]
+pub use block_cipher;
+#[cfg(feature = "block-cipher-trait")]
+mod block_cipher_trait {
+    use super::*;
+
+    use core::convert::TryFrom;
+
+    use block_cipher::generic_array::{
+        typenum::{self, Unsigned},
+        ArrayLength,
+    };
+    use block_cipher::{self, Block, BlockCipher, Key, NewBlockCipher};
+
+    impl KeyBits for typenum::U128 {
```

Comment:
In theory this can be implemented for any `T: typenum::Unsized` - do you want me to do that? I didn't do that for brevity in `impl BlockCipher for super::Key` (we don't have to specify other `typenum` bounds because we don't need `PartialDiv` in `type KeySize = ...` like we do in `NewBlockCipher`) but we could reduce duplication here.

Reply:
I suspect that the other algorithms (such as RC2) will require a lot more of `typenum`, so we should probably implement it for `T: typenum::Unsized` here.

Reply:
Done in ce9e11c45547212877629dea8f4007bd26501216